modules/cudaimgproc/include/opencv2/cudaimgproc.hpp

```diff
@@ -588,6 +588,22 @@ CV_EXPORTS Ptr<CornersDetector> createGoodFeaturesToTrackDetector(int srcType, i
 
 //! @} cudaimgproc_feature
 
+
+///////////////////////////// Median Filtering //////////////////////////////
+
+/** @brief Performs median filtering for each point of the source image.
+
+@param src Source image. Only CV_8UC1 images are supported for now.
+@param dst Destination image the filtered image. It has the same size and type
+as src .
+@param kernel Size of the kernerl used for the filtering.
+@param partition Specifies the parallel granularity of the workload. This parameter should be used GPU experts when optimizing performance.
+
+Outputs an image that has been filtered using median-filtering formulation. 
+ */
+CV_EXPORTS void medianFiltering(InputArray src, OutputArray dst, int kernel, int partition=128);
```

Comment:
Maybe it will be better to move this function to `cudafilters` module and implement it as a `cv::cuda::Filter`?